pkg/config/config.go

```diff
@@ -0,0 +1,42 @@
+package config
+
+import (
+	"path/filepath"
+
+	"github.com/fusor/cpma/pkg/env"
+	"github.com/fusor/cpma/pkg/io"
+	"github.com/sirupsen/logrus"
+)
+
+// reference:
+// https://docs.openshift.com/container-platform/3.11/install_config/master_node_configuration.html
+
+// Config contains CPMA configuration information
+type Config struct {
+	OutputDir string
+	Hostname  string
+}
+
+// Fetch files from the OCP3 cluster
+func (c *Config) Fetch(path string) ([]byte, error) {
+	dst := filepath.Join(c.OutputDir, c.Hostname, path)
+	logrus.Infof("Fetching file: %s", dst)
+	f, err := io.GetFile(c.Hostname, path, dst)
+	if err != nil {
+		return nil, err
+	}
+	logrus.Infof("File:loaded: %v", dst)
+
+	return f, nil
+}
+
+// LoadConfig collects and stores configuration for CPMA
+func LoadConfig() Config {
+	logrus.Info("Loaded config")
+
+	config := Config{}
+	config.OutputDir = env.Config().GetString("OutputDir")
+	config.Hostname = env.Config().GetString("Source")
```

Comment:
I prefer specifying the fields during the `{}`:
```golang
config := Config{
    OutputDir: env.Config().GetString("OutputDir")
    Hostname: env.Config().GetString("Source")
}
```

Actually I'd probably go a step further and replace the return with:

```golang
return  Config{
    OutputDir: env.Config().GetString("OutputDir")
    Hostname: env.Config().GetString("Source")
}
```

That might be going to far for some but at a minimum I'd use the first form of creating the instance.

Reply:
I appreciate the style difference.
And I'm happy to use the fields during the {}.
As long as we're consistent across the project.

That said I don't like using it for readability reason, especially when there are embedded struct(s).
I'm extrapolating, the good practice of defining structs separately instead of embedded sub structs, and apply it to assignments of as well.
And so, no matter if an object has or not sub structs, in order to be consistent,  use struct method for assignments.

Reply:
I suppose depending on the context one form might better than the other.
Anyway I've changed it to be consistent and also because with the return it's more concise.